support-workers/src/main/scala/com/gu/emailservices/SupporterPlusEmailFields.scala

```diff
@@ -0,0 +1,81 @@
+package com.gu.emailservices
+
+import com.gu.emailservices.SubscriptionEmailFieldHelpers.{formatDate, hyphenate, mask}
+import com.gu.salesforce.Salesforce.SfContactId
+import com.gu.support.workers._
+import com.gu.support.workers.states.SendThankYouEmailState.{
+  SendThankYouEmailContributionState,
+  SendThankYouEmailSupportertPlusState,
+}
+import org.joda.time.DateTime
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class SupporterPlusEmailFields(
+    getMandate: String => Future[Option[String]],
+    created: DateTime,
+) {
+
+  def build(
+      state: SendThankYouEmailSupportertPlusState,
+  )(implicit ec: ExecutionContext): Future[EmailFields] = {
+    getPaymentFields(
+      state.paymentMethod,
+      state.accountNumber,
+      created,
+    ).map { paymentFields =>
+      val fields = List(
+        "EmailAddress" -> state.user.primaryEmailAddress,
+        "created" -> created.toString,
+        "amount" -> state.product.amount.toString,
+        "currency" -> state.product.currency.iso,
+        "edition" -> state.user.billingAddress.country.alpha2,
+        "name" -> state.user.firstName,
+        "product" -> s"${state.product.billingPeriod.toString.toLowerCase}-contribution",
```

Comment:
is it supposed to be `contribution` here? Same question for line 37

Reply:
Well spotted, this is all a bit up in the air at the moment but that was definitely wrong